cmd/rpc-server.go

```diff
@@ -0,0 +1,67 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/rpc"
+
+	http2 "github.com/minio/minio/pkg/http"
+)
+
+// Can connect to RPC service using HTTP CONNECT to rpcPath.
+const rpcConnectReply = "200 Connected to Go RPC"
+
+func (server *rpcServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodConnect {
+		w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		io.WriteString(w, fmt.Sprintf("405 must %s\n", http.MethodConnect))
+		return
+	}
+
+	conn, _, err := w.(http.Hijacker).Hijack()
+	if err != nil {
+		errorIf(err, "rpc hijacking: %s", req.RemoteAddr)
+		return
+	}
+
+	// Overrides Read/Write deadlines if any.
+	conn = newRPCConn(conn)
+
+	io.WriteString(conn, "HTTP/1.0 "+rpcConnectReply+"\n\n")
+	server.ServeConn(conn)
+}
+
+type rpcServer struct {
+	*rpc.Server
```

Comment:
If `rpcServer` extends `rpc.Server`, then `newRPCServer()` is simply `return &rpcServer{}`